### PR TITLE
chore: Update Edge Proxy docs

### PR DESCRIPTION
**Pull request by khvn26**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

This PR adds a complete Edge Proxy settings reference. 

## How did you test this code?

Observed the docs preview deployment.

### CODE REVIEW

File: docs/docs/deployment/hosting/locally-edge-proxy.md [hunk 1]
```diff
@@ -10,14 +10,129 @@ The Edge Proxy can be configured using a json configuration file (named `config.
...
+  "server_side_key": "your_server_side_key",
+  "client_side_key": "your_client_side_environment_key"
+}]
+```
+
+### `api_poll_frequency`
+
+:::note
+
+This setting is optional.
```
Comment (matthewelwell) [inline docs/docs/deployment/hosting/locally-edge-proxy.md:28]: We should specify what the default is here.

File: docs/docs/deployment/hosting/locally-edge-proxy.md [hunk 2]
```diff
@@ -10,14 +10,129 @@ The Edge Proxy can be configured using a json configuration file (named `config.
...
+
+```json
+"endpoint_caches": {
+  "flags": {
+    "use_cache": false
+  },
+  "identities": {
+    "use_cache": false
+  }
+}
```
Comment (matthewelwell) [inline docs/docs/deployment/hosting/locally-edge-proxy.md]:
```suggestion
"endpoint_caches": {
  "flags": {
    "use_cache": false,
    "cache_max_size": 128,
  },
  "identities": {
    "use_cache": false,
    "cache_max_size": 128
  }
}
```

We can then also just include that the `cache_max_size` refers to the number of items in the cache before the oldest is evicted to find space for a new item.

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: docs/docs/deployment/hosting/locally-edge-proxy.md (hunk 1 above)
Comment (khvn26) [reply docs/docs/deployment/hosting/locally-edge-proxy.md:28]: Nice catch, fixed.

Comment (khvn26) [reply docs/docs/deployment/hosting/locally-edge-proxy.md:28]: Also, added `api_poll_timeout`.

File: docs/docs/deployment/hosting/locally-edge-proxy.md (hunk 2 above)
Comment (khvn26) [reply docs/docs/deployment/hosting/locally-edge-proxy.md]: Done.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**